### DISCUSSION TITLE
Change location of dependency crates

### DISCUSSION THREAD

**Discussion by rveenker** [alire-project/alire#1416] [Q&A]
In previous versions of Alire, the dependency crates were stored under the 'client' directory 'alire/cache/dependencies'.
More recent versions store these crates under the local user directories, which gives the advantage of not having a lot of copies of a crate on your system.

I would like to be able to choose where these dependency crates are stored, because having copy of a crate per client means I could make small changes to the contents of the crate (for testing/debugging) without having to go through the publishing / updating cycle. 
This way the original crate stays the same and does not affect other users of the crate.

My suggestion would be to add an option in the client manifest 'alire.toml' where you can specify if you want a local copy of a crate in a subdirectory or not.
Note that there is something similar already done for pinning.

BTW, I also tried changing the path using the config variable 'dependencies.dir' but could not get it to work.

**Comment by mosteo** [comment]
Hi Rob, I'm working on what you allude to, but at present things are as always yet?

Nonetheless, the working idea is just as you describe: there will be a config option so you can have dependencies just as before under `./alire`. 

Note that binary dependencies have always been downloaded to a common location outside of workspaces; this location was moved from `.config` to `.cache`.
